server.go

```diff
@@ -1097,10 +1103,25 @@ func (s *server) peerConnected(conn net.Conn, connReq *connmgr.ConnReq,
 	inbound bool) {
 
 	brontideConn := conn.(*brontide.Conn)
-	peerAddr := &lnwire.NetAddress{
-		IdentityKey: brontideConn.RemotePub(),
-		Address:     conn.RemoteAddr().(*net.TCPAddr),
-		ChainNet:    activeNetParams.Net,
+	var peerAddr *lnwire.NetAddress
+	if _, ok := cfg.net.(*RegularNet); ok {
```

Comment:
I would refactor this slightly:
```go
var addr *net.TCPAddr
switch remote := conn.RemoteAddr().(type){
  case *net.TCPAddr:
    addr = remote
  case *socks.ProxiedAddr:
    addr = &net.TCPAddr{
      IP:   net.ParseIP(remote.Host),
      Port: remot.Port,
    }
  default:
    return error
}

peerAddr = &lnwire.NetAddress{	 
  IdentityKey: brontideConn.RemotePub(),
  Address:     addr,
  ChainNet:    activeNetParams.Net,
}
```

Reply:
Alos wondering if this can be made even more abstract by changing `NetAddress.Address` to be of type `Addr` instead of `TCPAddr` so it can take the `RemoteAddr()` directly 🤔

Reply:
I think you're right that `NetAddress.Address` should take an `Addr`